File: myscripts/python/testscripts/matrixmult.py
import socket
import sys

total = 0
buffer_size = 1000

matrix_array = [[(i*buffer_size + j) for j in range(buffer_size)] for i in range(buffer_size)]

matrix_prod = [[sum(a*b for a,b in zip(i,j)) for j in zip(*matrix_array)] for i in matrix_array]

print matrix_prod

if len (sys.argv) > 1:
    portno = int(sys.argv[1])
    # Create a TCP/IP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # Connect the socket to the port where the server is listening
    server_address = ("localhost", portno)
    print >>sys.stderr, 'connecting to %s port %s' % server_address
    sock.connect(server_address)

    try:
        # Send data
        message = bytearray([1 for b in range(buffer_size) ])
        print >>sys.stderr, 'sending %d bytes' % len(message)
        sock.sendall(message)

        total += len(message)
        print >>sys.stderr, 'sent %d total bytes' % total

    finally:
        print >>sys.stderr, 'closing socket'
        sock.close()

File: myscripts/python/testscripts/MatrixMult.java
import java.io.*;
import java.net.*;

class MatrixMult {
    public static void main(String argv[]) throws Exception {
        int i, j, k;
        int portno;
        int buffer_size = 1000;
        int total = 0;

        int[][] matrix_array = new int[buffer_size][buffer_size];
        int[][] matrix_prod = new int[buffer_size][buffer_size];

        for (i = 0; i < buffer_size; ++i) {
                for (j = 0; j < buffer_size; ++j) {
                        matrix_array[i][j] = i*buffer_size + j;
                        matrix_prod[i][j] = 0;
                }
        }

        for (i = 0; i < buffer_size; ++i) {
                for (j = 0; j < buffer_size; ++j) {
                        for (k = 0; k < buffer_size; ++k) {
                                matrix_prod[i][j] += matrix_array[i][k]*matrix_array[k][j];
                        }
                }
        }

        for (i = 0; i < buffer_size; ++i) {
                for (j = 0; j < buffer_size; ++j) {
                        System.out.print(matrix_prod[i][j]+"\t");
                }
                System.out.print("\n");
        }

        if (argv.length > 0) {
        	try {
                        portno = Integer.parseInt(argv[0]);
	                Socket clientSocket = new Socket("localhost", portno);
	                byte[] buf = new byte[buffer_size];
	                DataOutputStream outToServer = new DataOutputStream(clientSocket.getOutputStream());
	                outToServer.write(buf);
	                System.out.println("Sent " + buffer_size + " bytes\n");
	                total += buffer_size;
	                clientSocket.close ();
			System.out.println("Sent " + total + " bytes\n");
                } catch (NumberFormatException e) {
                    System.err.println("Argument " + argv[1] + " must be an integer.");
                    System.exit(1);
                }
        }
    }
}
